vidarr-server/src/test/java/ca/on/oicr/gsi/vidarr/server/MainIntegrationTest.java

```diff
@@ -0,0 +1,808 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import org.flywaydb.core.Flyway;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.postgresql.ds.PGSimpleDataSource;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+public class MainIntegrationTest {
+  @ClassRule
+  public static JdbcDatabaseContainer pg =
+      new PostgreSQLContainer("postgres:12-alpine")
+          .withDatabaseName("vidarr-test")
+          .withUsername("vidarr-test")
+          .withPassword("vidarr-test");
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static ServerConfiguration config;
+  private static Main main;
+  private static HttpClient CLIENT =
+      HttpClient.newBuilder().followRedirects(Redirect.NORMAL).build();
+
+  private static ServerConfiguration getTestServerConfig(GenericContainer pg) {
+    ServerConfiguration config = new ServerConfiguration();
+    config.setName("vidarr-test");
+    config.setDbHost(pg.getHost());
+    config.setDbName("vidarr-test");
+    config.setDbPass("vidarr-test");
+    config.setDbUser("vidarr-test");
+    config.setDbPort(pg.getFirstMappedPort());
+    config.setPort(8999);
+    config.setUrl("http://localhost:" + config.getPort());
+    config.setOtherServers(new HashMap<>());
+    config.setInputProvisioners(Collections.singletonMap("raw", new RawInputProvisioner()));
+    config.setWorkflowEngines(new HashMap<>());
+    config.setOutputProvisioners(new HashMap<>());
+    config.setRuntimeProvisioners(new HashMap<>());
+    config.setTargets(new HashMap<>());
+    return config;
+  }
+
+  @BeforeClass
+  public static void setup() throws SQLException {
+    config = getTestServerConfig(pg);
+    main = new Main(config, "TEST");
+    main.startServer(main);
+    RestAssured.baseURI = config.getUrl();
+    RestAssured.port = config.getPort();
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    defaultParser = Parser.TEXT;
+    MAPPER.registerModule(new JavaTimeModule());
+  }
+
+  @Before
+  public void cleanAndMigrateDB() {
+    final var simpleConnection = new PGSimpleDataSource();
+    simpleConnection.setServerNames(new String[] {config.getDbHost()});
+    simpleConnection.setPortNumbers(new int[] {config.getDbPort()});
+    simpleConnection.setDatabaseName(config.getDbName());
+    simpleConnection.setUser(config.getDbUser());
+    simpleConnection.setPassword(config.getDbPass());
+    var fw = Flyway.configure().dataSource(simpleConnection);
+    fw.load().clean();
+    //    fw.locations("classpath:db/migration").load().migrate();
```

Comment:
This commented out code got moved, right?

Reply:
I had moved it to Main, but it makes more sense to move it back here so that Main doesn't need to know about test resources. Will update.

Reply:
Oh, and this allows me to remove that `Env` enum. Even better. Fix pushed.